src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Hosting/VSInternalServerCapabilitiesExtensions.cs

```diff
@@ -65,4 +67,21 @@ public static void EnableMapCodeProvider(this VSInternalServerCapabilities serve
     {
         serverCapabilities.MapCodeProvider = true;
     }
+
+    public static void EnableOnAutoInsert(
+        this VSInternalServerCapabilities serverCapabilities,
+        IEnumerable<string> triggerCharacters)
+    {
+        serverCapabilities.OnAutoInsertProvider = new VSInternalDocumentOnAutoInsertOptions()
+            .EnableOnAutoInsert(triggerCharacters);
+    }
+
+    public static VSInternalDocumentOnAutoInsertOptions EnableOnAutoInsert(
```

Comment:
Please don't add this to an unrelated type. If this extension method is needed, please create a new `VSInternalDocumentOnAutoInsertOptionsExtensions` class to place it on rather than putting it on an extensions class intended for `VSInternalServerCapabilities`.

Reply:
In Alex's defence this is an established pattern we have to deal with the pain of static capabilities in our LSP server, and dynamic capabilities in Cohosting. See semantic tokens above, in this file.

Reply:
Renamed per suggestion